aten/src/ATen/native/native_functions.yaml

```diff
@@ -1917,6 +1919,7 @@
 
 - func: _index_put_impl_(Tensor(a!) self, Tensor?[] indices, Tensor values, bool accumulate=False, bool unsafe=False) -> Tensor(a!)
   variants: function
+  device_guard: False
```

Comment:
Allow indices on CPU. Device guard is added in `TensorAdvancedIndexing.cpp`.

Reply:
Maybe these comments should be durably recorded on `native_functions.yaml` directly